### DISCUSSION TITLE
Storedge battery control best practices

### DISCUSSION THREAD

**Discussion by frank8the9tank** [WillCodeForCats/solaredge-modbus-multi#513] [General]
Hello all,

I would like to better understand the battery control options of the storedge inverter that i want to control.

The current setup is a Solaredge SE10K-RWS48BEN4 storedge inverter, with 2 Seplos 15kwh batteries connected.
The Seplos batteries talk over CAN with the pylon protocol to the inverter. which looks to be working fine.

What i try to understand the is remote control options that the integration has.
I would like to just have a single point of control. so a single box with a range of -5000 to 5000, (charging and discharging)
but the integration is net setup this way, so i do:

Discharge to maximise export, and them discharge limit value (0-5000)
and
Charge from grid and solar, and then set the charge limit value. (0-5000)
This way i can manually charge and discharge the battery.

There is a lifetime command timeout, which if set to 3600s, will run a "command" for a hour then jumps back to the default control mode.

Some questions i have are now:

1, When we have a lifetime command timeout, how do i re-trigger this timer? before the control mode jumps back to default?
2, There is a refresh button, but what does this do?
3, The voltage value seen off when i enable the battery, i see ~800V, but normal solaredge voltage is 750VDC, and 48VDC for battery,
So looks almost that these values are added in the integration or inverter. it's the same for the B1 voltage, which should be 48VDC but is ~800VDC.
4, For some reason the remote control just stops working, within the lifetime timeout, could it be that the connection to the inverter is lost? as i seen the the logbook of HA that around this time the entities where unavailable.
5, i see a lot of time that the default value for charge and discharge limit are set back. i think this is a bad practice as these are the maximum value for the system. for example I'm charging with 1kW, but then the value reset, and nog I'm charging with 5kW. which is not relay safe, as this happens uncontrolled.

can somebody explain these things?

**Comment by mpredfearn** [comment]
@frank8the9tank probably a good start would be to read the Solaredge documentation in this repository https://github.com/WillCodeForCats/solaredge-modbus-multi/blob/main/doc/Power-Control-Open-Protocol-for-SolarEdge-Inverters.pdf (page 16 onwards) so that you understand the controls as Solaredge expose them. These are coupled directly to the controls of the integration. The document will answer a lot of your questions.

Personally, I use the "default" storage controls to change the behaviour of my inverter / battery, so that I don't have the "fallback to default behaviour after a timeout" issue.

**Comment by frank8the9tank** [comment]
@mpredfearn i have seen and read this, but this does not anwser why the default values sometimes comes back and the voltage values are "strange"

**Reply by mpredfearn** [reply]
The battery voltage is read directly from the battery register 0xE170 (https://github.com/WillCodeForCats/solaredge-modbus-multi/blob/main/custom_components/solaredge_modbus_multi/hub.py#L1570, https://github.com/WillCodeForCats/solaredge-modbus-multi/blob/main/custom_components/solaredge_modbus_multi/sensor.py#L1726) so if it's "strange" that can only be coming from the inverter. Have you tried using a different modbus client (e.g. https://github.com/nmakel/solaredge_modbus) to explore the modbus registers and see if the value is also wrong there?
I don't why the default values come back on your inverter - mine do not though mine is an older inverter (SE3680) with older firmware so perhaps something has changed here. Maybe some other control interface (e.g. the Solaredge app via the cloud) is also changing these registers?

**Comment by WillCodeForCats** [comment]
The integration exposes the controls the inverter provides and the inverter itself does not provide a simplified "single point of control". When the SE website or app may provide simplified controls they are interpreting other controls. Ideally what someone could do is make some templates or scripts or whatever to provide simplified controls for Home Assistant, but it won't be me as the integration author because I don't have batteries to work with.

If you are setting "strange" values you can download your diagnostic file to see the raw values. Download it at the point in time where you see "normal" values too for comparison and upload both files. Or turn on debug mode but that can generate a lot of data depending on what your interval is set to.

I have seen other reports about the inverter resetting to default values because apparently that's just something they will do. Suggestions on how to handle that are here:

https://community.home-assistant.io/t/solaredge-modbus-configuration-for-single-inverter-and-battery/464084/582

**Comment by frank8the9tank** [comment]
I have added this integration twice to HA, I presume that this is fine? ( i have 2 solaredge inverters with multiple devices that are connected, but not to each other)

**Reply by WillCodeForCats** [reply]
There's no limit to the number of hubs you can add.

**Comment by deqw** [comment]
> 1, When we have a lifetime command timeout, how do i re-trigger this timer? before the control mode jumps back to default?

Where you say "control mode jumps back" did you mean "Command Mode jumps back"?

Do you have "Storage Control Mode" set at "Remote Control"? Is it in that mode all the time?

**Comment by frank8the9tank** [comment]
Yes i mean storage command mode, which jump back to the storage default mode.
i also see the following behaviour, the power goes to zero at some points:
![afbeelding](https://github.com/WillCodeForCats/solaredge-modbus-multi/assets/84384888/06e55cad-8818-42f2-954e-3fef224cbd92)
What could cause this? i keep setting the entities every 10 seconds trough an automation.

i think that also lose connection to the inverter:
![afbeelding](https://github.com/WillCodeForCats/solaredge-modbus-multi/assets/84384888/8c00478d-e0a4-4f2c-9986-21f693ba27e7)
as it shows unavailable some times. but my other solar edge inverter does not show this behaviour

**Comment by deqw** [comment]
The two inverters have a different modbus address?

I have Storage Control Mode always set at "Remote Control"; no changes. Storage Command Mode is most of the time set at "Maximize Self Consumption". I don't do discharge to the grid.

What is your Battery Mode? See the SolarEdge app Battery Mode Setting. When you do all the settings yourself anyway you can try "Revert to Latest Installer Settings" and see what happens from there on.

> as it shows unavailable some times. but my other solar edge inverter does not show this behaviour
I see "unavailable too from time to time" which recovers in a few seconds.

**Reply by frank8the9tank** [reply]
I have added the integration two times to home assistant, as both system are not connected to each other.
The battery mode is controlled by HA, i dont use the solaredge app. i want to control it locally.
The main issue at the moment is that the inverter "stop" at random points.
